.github/workflows/synth-test.yml

```diff
@@ -0,0 +1,29 @@
+
+name: synth-test
+
+on:
+  workflow_call:
+    inputs:
+      CONTEXT:
+        required: true
+        type: string
+
+jobs:
+  tests:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      - name: cdk synth
+        uses: youyo/aws-cdk-github-actions@v2
+        with:
+          cdk_subcommand: 'synth'
+          actions_comment: false
+          cdk_args: '--context env=${{ inputs.CONTEXT }} --debug'
+
+      - name: Install dependencies
+        run: pip install -r requirements.txt -r requirements-dev.txt
+
+      - name: Run unit tests
+        run: python -m pytest tests/ -s -v
```

Comment:
you could move unit tests into `tests:` in main.yaml  since it doesn't depend on any inputs.

Reply:
Good point.  I have updated the PR accordingly.